src/mod_ir_temp.rs

```diff
@@ -0,0 +1,278 @@
+/// For instance, if 'THRESHOLD_AMBIENT' < 'ambient_temp', then 'ChangedToAmbientToHot' is triggered.
+use crate::config::Config;
+use i2cdev::linux::LinuxI2CError;
+use linux_embedded_hal::{Delay, I2cdev};
+use mlx9061x::ic::Mlx90614;
+use mlx9061x::{Error, Mlx9061x, SlaveAddr};
+
+const THRESHOLD_AMBIENT: f32 = 22.0;
+const THRESHOLD_OBJECT: f32 = 44.0;
+
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum IrTempStateChange {
+	None,
+	ChangedToAmbientToHot,
+	ChangedToObjectToHot,
+	ChanedToBothToHot,
+	ChangedToCancelled,
+}
+
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum IrTempState {
+	Normal,
+	TooHot,
+}
+
+pub struct ModIR {
+	mlx: Option<Mlx9061x<I2cdev, Mlx90614>>,
+	device: String,
+	addr: SlaveAddr,
+	pub ambient_temp: f32,
+	pub object_temp: f32,
+	_emissivity: f32,
+	active_ambient_state: IrTempState,
+	active_object_state: IrTempState,
+	data_interval: u16,
+	read_counter: u16,
+}
+
+impl ModIR {
+	/// This function is used to initialize a default instance.
+	pub fn new_default() -> Self {
+		Self {
+			mlx: None,
+			device: "/dev/null".to_string(),
+			addr: SlaveAddr::Default,
+			ambient_temp: 0.0,
+			object_temp: 0.0,
+			_emissivity: 1.0,
+			active_ambient_state: IrTempState::Normal,
+			active_object_state: IrTempState::Normal,
+			data_interval: 0,
+			read_counter: 0,
+		}
+	}
+
+	/// This function initializes the MOD-IR-TEMP and returns an instance of ModIR upon success.
+	/// In case of an error, the error code is returned.
+	pub fn new(config: &mut Config) -> Result<Self, Error<LinuxI2CError>> {
+		let mut s: Self;
+		if config.get_bool("ir/enable") {
+			s = Self {
+				mlx: None,
+				device: config.get::<String>("ir/device"),
+				addr: SlaveAddr::Default,
+				ambient_temp: 0.0,
+				object_temp: 0.0,
+				_emissivity: 1.0,
+				active_ambient_state: IrTempState::Normal,
+				active_object_state: IrTempState::Normal,
+				data_interval: config.get::<u16>("ir/data/interval"),
+				read_counter: config.get::<u16>("ir/data/interval"),
+			};
+			if s.device != "/dev/null" {
+				match s.init() {
+					Ok(_) => {
+						return Ok(s);
+					}
+					Err(error) => {
+						return Err(error);
+					}
+				}
+			}
+		} else {
+			s = ModIR::new_default();
+		}
+		Ok(s)
+	}
+
+	/// This function initiates the I2C connection to the MOD-IR-TEMP sensor.
+	/// It returns nothing on success and returns an error message on failure.
+	fn init(&mut self) -> Result<(), Error<LinuxI2CError>> {
+		match Mlx9061x::new_mlx90614(I2cdev::new(&self.device).unwrap(), self.addr, 5) {
+			Ok(mlx_sensor) => {
+				self.mlx = Some(mlx_sensor);
+				()
+			}
+			Err(error) => {
+				return Err(error);
+			}
+		}
+		Ok(())
+	}
+
+	/// This function checks if a state change has occurred and returns the corresponding state change.
+	/// It returns 'None' if nothing has changed since the last duration.
+	/// It returns 'ChangedToBothToHot' if both thresholds have been exceeded.
```

Comment:
Some o missing (also in other lines):

```suggestion
	/// It returns 'ChangedToBothTooHot' if both thresholds have been exceeded.
```